libkbfs/disk_block_cache.go

```diff
@@ -411,29 +406,59 @@ func (cache *DiskBlockCacheStandard) Put(ctx context.Context, tlfID tlf.ID,
 		return err
 	}
 	if !hasKey {
-		for cache.currBytes+uint64(encodedLen) > cache.maxBytes {
-			cache.evictLocked(ctx, defaultNumBlocksToEvict)
+		for {
+			select {
+			// Ensure we don't loop infinitely
+			case <-ctx.Done():
+				return ctx.Err()
+			default:
+			}
+			bytesAvailable, err := cache.limiter.beforeDiskBlockCachePut(ctx,
+				encodedLen)
+			if err != nil {
+				cache.log.CWarningf(ctx, "Error obtaining space for the disk"+
+					" block cache: %+v", err)
+				return err
+			}
+			if bytesAvailable >= 0 {
+				break
+			}
+			_, _, err = cache.evictLocked(ctx, defaultNumBlocksToEvict)
+			if err != nil {
+				return err
```

Comment:
I don't understand -- it sounds like you're referring to the error returned by the `beforeDiskBlockCachePut()` call above. But if that succeeds, and the call to `evictLocked` fails, you need an `afterDiskBlockCachePut` call, right?